src/main/java/org/springframework/data/elasticsearch/core/SearchOperations.java

```diff
@@ -170,32 +202,203 @@ default long count(Query query, IndexCoordinates index) {
 	 * @param clazz The class of entity to retrieve.
 	 * @param index the index to run the query against
 	 * @return scrolled page result
+	 * @deprecated since 4.0, use {@link #searchScrollStart(long, Query, Class, IndexCoordinates)}.
+	 */
+	@Deprecated
+	default <T> ScrolledPage<T> startScroll(long scrollTimeInMillis, Query query, Class<T> clazz,
+			IndexCoordinates index) {
+		return (ScrolledPage<T>) SearchHitSupport
+				.unwrapSearchHits(searchScrollStart(scrollTimeInMillis, query, clazz, index));
+	}
+
+	/**
+	 * Returns next scrolled page.
+	 *
+	 * @param scrollId the scroll id
+	 * @param scrollTimeInMillis duration of the scroll time
+	 * @param clazz The class of entity to retrieve.
+	 * @return scrolled page result
+	 * @deprecated since 4.0, use {@link #searchScrollStart(long, Query, Class, IndexCoordinates)}.
 	 */
-	<T> ScrolledPage<T> startScroll(long scrollTimeInMillis, Query query, Class<T> clazz, IndexCoordinates index);
+	@Deprecated
+	default <T> ScrolledPage<T> continueScroll(@Nullable String scrollId, long scrollTimeInMillis, Class<T> clazz) {
+		return (ScrolledPage<T>) SearchHitSupport
+				.unwrapSearchHits(searchScrollContinue(scrollId, scrollTimeInMillis, clazz));
+	}
 
-	<T> ScrolledPage<T> continueScroll(@Nullable String scrollId, long scrollTimeInMillis, Class<T> clazz);
+	/**
+	 * Returns next scrolled page
+	 *
+	 * @param scrollId the scroll id
+	 * @param scrollTimeInMillis duration of the scroll time
+	 * @param clazz The class of entity to retrieve.
+	 * @return scrolled page result
+	 */
+	<T> ScrolledPage<SearchHit<T>> searchScrollContinue(@Nullable String scrollId, long scrollTimeInMillis,
```

Comment:
`startScroll`/`continueScroll` methods are rather internal methods and we expose a streaming functionality on top of these. Once we have `SearchSlice` in place, we should be good to just change the return types of `startScroll`/`continueScroll` so there's no need to have `searchScrollContinue`/`searchScrollClear`.